src/components/ReportActionItem/ReportPreview.js

```diff
@@ -123,6 +123,8 @@ function ReportPreview(props) {
     const transactionsWithReceipts = ReportUtils.getTransactionsWithReceipts(props.iouReportID);
     const numberOfScanningReceipts = _.filter(transactionsWithReceipts, (transaction) => TransactionUtils.isReceiptBeingScanned(transaction)).length;
     const hasReceipts = transactionsWithReceipts.length > 0;
+    const distanceRequestTransactions = ReportUtils.getDistanceRequestTransactions(props.iouReportID);
```

Comment:
My recommendation could be opinionated, but we're looping through transactions quite a few times.

1.  getTransactionsWithReceipts
2. getDistanceRequestTransactions

Should we be combining these method? Is the list going to be too long to delay the loop?

Reply:
The list shouldn't generally exceed 5-10 reports because once the IOU report is settled, it's closed and a new one will be created.

To my mind, creating a separate method that combines both checks is not worth the hustle: it's either creating a new function in ReportUtils that will be used in this only place or fetching all the report's transactions here and placing these custom calculations inside the `ReportPreview`, which will complicate the already quite cluttered code.

What do you think?

Reply:
That's true for users, but admins will likely see 10's of IOUs towards the end of the month. Though we probably don't need to improve this until it becomes a problem.